src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -201,7 +201,7 @@ public class SystemTransaction : Transaction { }
     /// <remarks>Created because of cyclic dependencies between Core and Rlp modules</remarks>
     public interface ITransactionSizeCalculator
     {
-        int GetLength(Transaction tx);
+        int GetNetworkTxLength(Transaction tx);
```

Comment:
I would keep it as `GetLength` and then have a `NetworkTransactionSizeCalculator` that would take `TxDecoder` and call `GetLength(tx, RlpBehaviors.InMempoolForm);`

Then you can reuse this interface for other calculators if needed.